tests/test_other.py

```diff
@@ -1511,6 +1511,45 @@ def test_js_link(self):
     Popen([PYTHON, EMCC, os.path.join(self.get_dir(), 'main.cpp'), '--pre-js', 'before.js', '--post-js', 'after.js', '-s', 'NO_EXIT_RUNTIME=1']).communicate()
     self.assertContained('hello from main\nhello from js\n', run_js(os.path.join(self.get_dir(), 'a.out.js')))
 
+  def test_emscripten_block_on(self):
+    open(os.path.join(self.get_dir(), 'main.c'), 'w').write('''
+      #include <stdio.h>
+      #include <emscripten.h>
+
+      #define INPUT_LENGTH 7
+
+      int main(int argc, char **argv) {
+        char input[INPUT_LENGTH] = {};
+        for (int n = 0; n < INPUT_LENGTH; ++n) {
+          emscripten_block_on("char_ready");
```

Comment:
Looking at this test case, it seems that `emscripten_block_on()` is a public interface function that users will call directly in their code.